src/EVM/Expr.hs

```diff
@@ -478,6 +454,54 @@ minLength bufEnv = go 0
       b <- Map.lookup a bufEnv
       go l b
 
+-- returns the largest prefix that is guaranteed to be concrete (if one exists)
+concretePrefix :: Expr Buf -> Maybe (Vector (Expr Byte))
+concretePrefix b = let v = mkvec in case V.length v of
+  0 -> Nothing
```

Comment:
If the input is an empty concrete buffer the output will be `Nothing` and I'm not sure if this is the correct semantics (empty vector would make more sense). I don't know if we will ever call this with an empty buffer though since we use it with initcode.